### PR TITLE
Added Developer Vertical

### PR DESCRIPTION
**Pull request by dzadikdev**
Adding Dev Portal Functionality

### CODE REVIEW

File: GlobalBehaviorandComponents/manageapps.py [hunk 1]
```diff
@@ -23,13 +23,19 @@ def gbac_apps():
     okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
 
     applist = okta_admin.get_applications_by_user_id(user_info["sub"])
+    myapplist = []
+    for app in applist:
+        if "profile" in app:
+            if "createdby" in app["profile"]:
+                if user_info["email"] in app["profile"]["createdby"]:
+                    myapplist.append(app)
```
Comment (noinarisak) [inline GlobalBehaviorandComponents/manageapps.py:31]: Wouldn't this same logic work? Nested `if` are hard to read.

```
if ("profile" in app) AND ("createdby" in app["profile"]) AND (user_info["email"] in app["profile"]["createdby"]):
    myapplist.append(app)

File: GlobalBehaviorandComponents/manageapps.py [hunk 2]
```diff
@@ -87,12 +94,14 @@ def gbac_apps_delete():
...
+    okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
 
-    # app_name = request.args.get('appname')
-    # app_id = request.args.get('appid')
-    message = "Application Updated"
-    return redirect(url_for("gbac_manageapps_bp.gbac_apps", _external=True, _scheme=session[SESSION_INSTANCE_SETTINGS_KEY]["app_scheme"], message=message))
+    oidcclientid = request.args.get('oidcclientid')
+    oidcloginredirecturi = request.args.get('oidcloginredirecturi')
+    oidcapplabel = request.args.get('oidcapplabel')
+
```
Comment (noinarisak) [inline GlobalBehaviorandComponents/manageapps.py:102]: Not checking if `oidc_clientid`, `oidc_login_redirect_uri`, `oidc_app_label` are None. What happens when the variables are None or ""?

File: GlobalBehaviorandComponents/manageapps.py [hunk 3]
```diff
@@ -87,12 +94,14 @@ def gbac_apps_delete():
...
+    okta_admin.update_web_application(app_label=oidcapplabel, redirect_uris=oidcloginredirecturi, app_id=oidcclientid)
+    return ""
```
Comment (noinarisak) [inline GlobalBehaviorandComponents/manageapps.py:104]: Just `return` remove `""`, unless you don't need it at all.

File: GlobalBehaviorandComponents/managetrustedorigins.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+@gbac_managetrustedorigins_bp.route("/createtrustedoriginsfinal")
+@apply_remote_config
+@is_authenticated
+def gbac_trustedorigins_create_final():
+    logger.debug("gbac_trustedorigins_create_final()")
+    user_info = get_userinfo()
+    okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
+    trustedoriginurl = request.args.get('trustedoriginurl')
+    trustedoriginid = request.args.get('id')
```
Comment (noinarisak) [inline GlobalBehaviorandComponents/managetrustedorigins.py:107]: Not checking if variables exist. What happens when it `None` or empty?

File: _developer/templates/developer/template.html [hunk 5]
```diff
@@ -1,9 +1,16 @@
-<!--Do Not Remove any parts of this document unless you expect to not use out of the box behaviors-->
+<!--Do Not Remove any parts of this document unless you expect to not use out of the box behaviours-->
```
Comment (noinarisak) [inline _developer/templates/developer/template.html:1]: change to "behaviors" is this Canada? ;)

File: utils/okta.py [hunk 6]
```diff
@@ -606,8 +630,183 @@ def create_linked_users(self, userid, parentid, name):
 
         return RestUtil.execute_put(url=url, body=None, headers=okta_headers)
 
-    def create_clientcredential_application(self, app_name):
-        self.logger.debug("OktaAdmin.get_applications_by_id(app_id)")
+    def create_auth_server(self, auth_server_name, app_server_description):
```
Comment (noinarisak) [inline utils/okta.py:633]: So the Okta Python SDK couldn't help here? I thinking it would reduce the code count.

**Review by noinarisak — CHANGES_REQUESTED** [review]
@dzadikdev Some items to look over.

File: GlobalBehaviorandComponents/manageapps.py (hunk 1 above)
Comment (dzadikdev) [reply GlobalBehaviorandComponents/manageapps.py:31]: It worked.

File: GlobalBehaviorandComponents/manageapps.py (hunk 2 above)
Comment (dzadikdev) [reply GlobalBehaviorandComponents/manageapps.py:102]: Fixed.

File: GlobalBehaviorandComponents/manageapps.py (hunk 3 above)
Comment (dzadikdev) [reply GlobalBehaviorandComponents/manageapps.py:104]: Added, HTTP Response code.

File: _developer/templates/developer/template.html (hunk 5 above)
Comment (dzadikdev) [reply _developer/templates/developer/template.html:1]: Fine. Fixed. LOL

File: GlobalBehaviorandComponents/managetrustedorigins.py (hunk 4 above)
Comment (dzadikdev) [reply GlobalBehaviorandComponents/managetrustedorigins.py:107]: Added check and send http error code if empty.

File: utils/okta.py (hunk 6 above)
Comment (dzadikdev) [reply utils/okta.py:633]: This would require a complete rework and weeks of testing/fixing.

Comment (noinarisak) [reply utils/okta.py:633]: Gotcha. Okay.

**Comment by noinarisak** [comment]
LGTM!

**Review by noinarisak — APPROVED** [review]
LGTM

**Pull request merged**